planner: update document about multi-valued indexes (#14543)
pingcap/docs: choose-index.md

For the limitations of multi-valued indexes, refer to [`CREATE INDEX`](/sql-statements/sql-statement-create-index.md#limitations).

> ### Scenarios that multi-valued indexes can support

Reviewer: ```suggestion
### Supported scenarios
```

6 rows in set, 1 warning (0.00 sec)
```

> For `OR` conditions composed of multiple `member of` expressions, and these expressions can be used to access the same multi-valued index, then IndexMerge can be used:

Reviewer: ```suggestion
For `OR` conditions composed of multiple `member of` expressions that can access the same multi-valued index, IndexMerge can be used to access the multi-valued index:
```

+---------------------------------+---------+-----------+---------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------+
```

> ### Scenarios that multi-valued indexes cannot support completely
> 
> For `AND` conditions composed of multiple expressions, and these expressions are corresponding to multiple different indexes, then only one multi-valued index can be used:

Reviewer: ```suggestion
### Partially supported scenarios

For `AND` conditions composed of multiple expressions that correspond to multiple different indexes, only one multi-valued index can be used to access:
```

|   ├─IndexRangeScan_6(Build)     | 10.00   | cop[tikv] | table:t, index:k1(cast(json_extract(`j1`, _utf8'$.path') as signed array)) | range:[1,1], keep order:false, stats:pseudo                                                                                                    |
|   └─Selection_8(Probe)          | 0.01    | cop[tikv] |                                                                            | eq(test.t.a, 3)                                                                                                                                |
|     └─TableRowIDScan_7          | 10.00   | cop[tikv] | table:t                                                                    | keep order:false, stats:pseudo                                                                                                                 |
+---------------------------------+---------+-----------+----------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------+
5 rows in set, 6 warnings (0.01 sec)
```

> Currently only one index can be used, instead of generating a plan like below that can use multiple indexes to access at the same time:
> 
> ```
>   Selection
>   └─IndexMerge
>     ├─IndexRangeScan(k1)
>     ├─IndexRangeScan(k2)
>     ├─IndexRangeScan(ka)
>     └─Selection
>       └─TableRowIDScan
> ```
> 
> ### Scenarios that multi-valued indexes cannot support
> 
> For `OR` conditions composed of multiple expressions, and these expressions are corresponding to multiple different indexes, then multi-valued indexes cannot be used:

Reviewer: ````suggestion
Currently, TiDB only supports using one index to access instead of generating the following plan that uses multiple indexes to access at the same time:

```
Selection
└─IndexMerge
  ├─IndexRangeScan(k1)
  ├─IndexRangeScan(k2)
  ├─IndexRangeScan(ka)
  └─Selection
    └─TableRowIDScan
```

### Unsupported scenarios

For `OR` conditions composed of multiple expressions that correspond to multiple different indexes, multi-valued indexes cannot be used:
````

+-------------------------+----------+-----------+---------------+---------------------------------------------------------------------------------------------+
| id                      | estRows  | task      | access object | operator info                                                                               |
+-------------------------+----------+-----------+---------------+---------------------------------------------------------------------------------------------+
| Selection_5             | 8000.00  | root      |               | or(json_memberof(cast(1, json BINARY), json_extract(test.t.j1, "$.path")), eq(test.t.a, 3)) |
| └─TableReader_7         | 10000.00 | root      |               | data:TableFullScan_6                                                                        |
|   └─TableFullScan_6     | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo                                                              |
+-------------------------+----------+-----------+---------------+---------------------------------------------------------------------------------------------+
3 rows in set, 3 warnings (0.00 sec)
```

> An workaround for the above scenario is to use `Union All` to rewrite your queries.

Reviewer: ```suggestion
A workaround for the preceding scenario is to rewrite the query using `Union All`:
```
